File: code-coaches/medium/snowballing-numbers/SnowballingNumbers.java
import java.util.Scanner;

public class SnowballingNumbers {

    public static boolean isEachNumberGreaterThanSum(int[] numbers) {
        int sum = 0;
        for (int i = 1; i < numbers.length; i++) {
            sum += numbers[i - 1];
            if (numbers[i] <= sum) {
                return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int numbersLength = scanner.nextInt();
        int[] numbers = new int[numbersLength];

        for (int i = 0; i < numbersLength; i++) {
            numbers[i] = scanner.nextInt();
        }

        System.out.println(isEachNumberGreaterThanSum(numbers));

        scanner.close();
    }

}

File: code-coaches/medium/snowballing-numbers/SnowballingNumbers.cs
using System;

class SnowballingNumbers
{
    static bool IsEachNumberGreaterThanSum(int[] numbers)
    {
        int sum = 0;
        for (int i = 1; i < numbers.Length; i++)
        {
            sum += numbers[i - 1];
            if (numbers[i] <= sum)
            {
                return false;
            }
        }
        return true;
    }

    static void Main(string[] args)
    {
        int numbersLength = Convert.ToInt32(Console.ReadLine());
        int[] numbers = new int[numbersLength];

        for (int i = 0; i < numbersLength; i++)
        {
            numbers[i] = int.Parse(Console.ReadLine());
        }

        bool result = IsEachNumberGreaterThanSum(numbers);
        Console.WriteLine(result ? "true" : "false");
    }
}
